### PR TITLE
RATEST-230:Use cucumber Examples Table to fetch  Patient name in findPatientSteps

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RATEST-230

### CODE REVIEW

**Comment by sherrif10** [comment]
cc @kdaud  Looking forward for your reviews

**Comment by sherrif10** [comment]
Hey @kdaud waiting on review. Thanks

File: src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java [hunk 1]
```diff
@@ -40,9 +40,9 @@ public void noPatients() {
 		assertNotNull(getElement(By.className("dataTables_empty")));
 	}
 
-	@And("User enters John Smith")
-	public void enterJohnSmith() {
-		findPatientPage.enterPatient("John Smith");
+	@And("User enters {string} patientName")
+	public void enterJohnSmith(String patientName) {
```
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java:44]: @sherrif10 what do you think about the method  name ?

File: src/test/resources/features/refapp-2.x/03-trackPatient/findPatient.feature [hunk 2]
```diff
@@ -12,7 +12,10 @@ Feature: Patient Search
...
-    And User enters John Smith
+    And User enters "<patientName>" patientName
     Then Search Page returns patients
     And User clicks on first patient
-    Then System loads patient dashboard
\ No newline at end of file
+    Then System loads patient dashboard
+     Examples:
+      | patientName  | 
+      | John Smith    |
```
Comment (kdaud) [inline src/test/resources/features/refapp-2.x/03-trackPatient/findPatient.feature:21]: Minor comment: I would adjust the last symbol(`I`) on the right hand side to be in vertical line with the one above it

Comment (sherrif10) [reply src/test/resources/features/refapp-2.x/03-trackPatient/findPatient.feature]: Resolved thnks

File: src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java (hunk 1 above)
Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java]: Sure Resolved thanks.

**Comment by sherrif10** [comment]
Hello @kdaud  waiting on your approval please. its blocking me .Thanks

File: src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java [hunk 3]
```diff
@@ -40,9 +40,9 @@ public void noPatients() {
 		assertNotNull(getElement(By.className("dataTables_empty")));
 	}
 
-	@And("User enters John Smith")
-	public void enterJohnSmith() {
-		findPatientPage.enterPatient("John Smith");
+	@And("User enters {string} patientName")
+	public void enterPatientName(String patientName) {
```
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java:44]: @sherrif10 why not giving a generic methodName to cater for searching a patient using patientIdentifier or patIentName ?

Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java:44]: That is already catered for and perhaps in steps class , this is was to improve the logical way of improving code using cucumber table.  There are number of different ways where this has been done , especially in the[ login. feature](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/src/test/resources/features/refapp-2.x/02-login/login.feature) as an example.

**Comment by sherrif10** [comment]
Thanks @kdaud  Am merging this since am in the middle of a ticket that consumes it and I badly need to finish it before tonight if possible.Thanks alot.

**Comment by kdaud** [comment]
@sherrif10 have you taken a look at https://ci.openmrs.org/browse/CONTRIB-QA-795 after merging the commit ?

**Comment by sherrif10** [comment]
Sure am looking into that apparently thanks

**Comment by kdaud** [comment]
> Sure am looking into that apparently 

Waiting for the report on your investigation !

**Comment by kdaud** [comment]
@sherrif10 any update ?

**Comment by sherrif10** [comment]
Am still trying to rebuild the bamboo ci  and probably find out why
it's not working as expected.

**Comment by sherrif10** [comment]
Hello @dkayiwa . Am seeing a revert of this commit and am not clearly understanding it clearly

**Comment by dkayiwa** [comment]
@sherrif10 feel free to reopen the pull request, have it reviewed and approved. Then i will merge it.

**Comment by sherrif10** [comment]
> feel free to reopen the pull request, have it reviewed and approved. Then i will merge it.

ohh ok thanks

**Pull request merged**